### PR TITLE
Fix locale assign

### LINKED ISSUE #11: Add locale filter
Currently, when you select the default `en` locale, you only see courses that are in `en`. But if you select the `pt` locale, you see courses in `pt` AND `en`. It's a good solution for now, not gonna lie. If users start complaining about more control over the courses locale, then it's time to build this locale filter. Low priority for now

### PR DESCRIPTION
**Pull request by kevinam99**
## Why is this change is being made?
To fix content shown by the selected locale

## How is this being accomplished?
By changing how locales are returned from the function `search_locale/1`

## How do you manually test this?
Just visit the courses page

Attempts to resolve https://github.com/danielbergholz/techschool.dev/issues/11

### CODE REVIEW

**Comment by danielbergholz** [comment]
Hey @kevinam99, thank you so much for your contribution!

So, my current strategy for the locale is the following: Most of my Brazilian friends know English, so if you choose the `pt` locale, the website will show you all the courses (English and Portuguese). However, it's extremely unlikely that an American knows Brazilian Portuguese, so for them I'm only showing the courses in English. This is what the following line of code is doing:

```elixir
defp search_locale(%{assigns: %{locale: locale}}) do
    case locale do
      @default_locale -> [@default_locale]
      _ -> [locale, @default_locale]
    end
  end
```

If the user selected the default locale (English) -> only show courses in English
If the user selects any other language (Portuguese) -> show courses in that language + the courses in English.

AFAIK, your PR changes this behavior to only show the courses for the selected language. So any Brazilian visiting the website would only see courses in Portuguese, not English as well.

TL;DR: This is the expected behavior, not a bug 😅

LMK your thoughts about this! 🤘

**Comment by kevinam99** [comment]
I figured that it might be required based on the linked issue. If this is expected behaviour and is not expected to change, we can leave things as they are.

**Comment by danielbergholz** [comment]
Sorry for not being super clear, I'm going to close this for now, but thanks anyway!
